src/scheduler.rs

```diff
@@ -349,6 +349,16 @@ async fn download_msgs(context: &Context, imap: &mut Imap) -> Result<()> {
 
     for msg_id in msg_ids {
         if let Err(err) = download_msg(context, msg_id, imap).await {
+            // Update download state to failure
+            // so it can be retried.
+            //
+            // On success update_download_state() is not needed
+            // as receive_imf() already
+            // set the state and emitted the event.
+            msg_id
+                .update_download_state(context, DownloadState::Failure)
+                .await?;
+
             warn!(context, "Failed to download message {msg_id}: {:#}.", err);
```

Comment:
Probably should be before the `update_download_state()` call, it's unlikely that it fails, but still